File: src/dorkbox/netUtil/jna/windows/structs/IP_ADAPTER_DNS_SERVER_ADDRESS_XP.java
package dorkbox.netUtil.jna.windows.structs;

import java.util.Arrays;
import java.util.List;

import com.sun.jna.Structure;

public
class IP_ADAPTER_DNS_SERVER_ADDRESS_XP extends Structure {
    public static
    class ByReference extends IP_ADAPTER_DNS_SERVER_ADDRESS_XP implements Structure.ByReference {}


    public int Length;
    public int Reserved;

    public IP_ADAPTER_DNS_SERVER_ADDRESS_XP.ByReference Next;
    public SOCKET_ADDRESS Address;

    @Override
    protected
    List<String> getFieldOrder() {
        return Arrays.asList("Length", "Reserved", "Next", "Address");
    }
}

File: src/dorkbox/netUtil/jna/windows/structs/IP_ADAPTER_DNS_SERVER_ADDRESS_XP.kt
package dorkbox.netUtil.jna.windows.structs

import com.sun.jna.Structure

open class IP_ADAPTER_DNS_SERVER_ADDRESS_XP : Structure() {
    class ByReference : IP_ADAPTER_DNS_SERVER_ADDRESS_XP(), Structure.ByReference

    @JvmField var Length = 0
    @JvmField var Reserved = 0
    @JvmField var Next: ByReference? = null
    @JvmField var Address: SOCKET_ADDRESS? = null

    override fun getFieldOrder(): List<String> {
        return listOf("Length", "Reserved", "Next", "Address")
    }
}
